### PR TITLE
Add @WithSpan Annotation documentation and update the README

### PR DESCRIPTION
**Pull request by flands**
* Move build and configure to top as getting started section
* Add manual instrumentation section
* Document `@WithSpan` annotation
* Move developer specific information to CONTRIBUTING.md
* Cleanup formatting and use consistent spacing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,105 +117,75 @@ to capture telemetry from a number of popular libraries and frameworks.
...
 
-## Building from source
-
-Build using Java 8:
-
-```gradle assemble```
-
-and then you can find the java agent artifact at `java-agent/build/lib/opentelemetry-auto-<version>.jar`.
+Note these logs are extremely verbose. Enable debug logging only when needed.
+Debug logging negatively impacts the performance of your application.
```
Comment (trask) [inline README.md:191]: 😄👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,107 +1,174 @@
...
+Pull requests for bug fixes are welcome, but before submitting new features
+or changes to current functionality [open an
+issue](https://github.com/open-telemetry/opentelemetry-auto-instr-java/issues/new)
+and discuss your ideas or propose the changes you wish to make. After a
+resolution is reached a PR can be submitted for review.
 
-In order to build and test this whole repository you need JDK 11+.
-Some instrumentations and tests may put constraints on which java versions they support.
-See [Executing tests with specific java version](#Executing tests with specific java version) below.
+In order to build and test this whole repository you need JDK 11+. Some
```
Comment (iNikem) [inline CONTRIBUTING.md:9]: @flands I very like the suggestion of Dan Allen from https://www.youtube.com/watch?v=r6RXRi5pBXg about "Write one sentence per line". It makes diffs and merges easier and don't change rendered layout in any way.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,107 +1,174 @@
...
+libs/exporter-support-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-exporters-jaeger-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-exporters-logging-<version>-SNAPSHOT.jar
+libs/opentelemetry-auto-exporters-otlp-<version>-SNAPSHOT.jar
+```
+
+#### Building from source
+
+Build using Java 8:
```
Comment (iNikem) [inline CONTRIBUTING.md:104]: You need java 11 to build current repo

File: CONTRIBUTING.md (hunk 2 above)
Comment (trask) [reply CONTRIBUTING.md:9]: oh, yes, the specification repo has been battling with this decision: https://github.com/open-telemetry/opentelemetry-specification/pull/564. let's see what they end up implementing, and use the same? seems like they are leaning in the direction u mention

File: CONTRIBUTING.md (hunk 3 above)
Comment (trask) [reply CONTRIBUTING.md:104]: good catch 👍, i can fix this tomorrow if u don't beat me to it

**Pull request merged**